### ISSUE TITLE
Google system apps don't trust adguard certificate

### ISSUE THREAD

**Issue by techIndia-hacker** [AdguardTeam/AdguardForAndroid#3530]
Google system apps don't trust adguard certificate if installed to root store using magisk module. But the certificate is trusted entirely but all Google apps only if adguard itself installs the certificate using root access in android 10.
Now in Samsung devices adguard can actually install root certificate in trusted store but in other devices like oneplus it fails. Oneplus devices needs the magisk module to install certificate in system store. Thus system apps like playstore, play services,  even Vanced YouTube doesn't trust adguard certificate in oneplus devices. 

Ideally though installation through magisk module should work in the way adguard does it so that adguard certificate is trusted system wide.

Test devices:- Samsung A71 Android 10, oneplus 8 pro Android 10.
Root access: both devices using magisk

**Comment by TheHasagi** [comment]
@techIndia-hacker 

Are you talking about installing AdGuard CA into the system store via magisk? It is working fine on my side. So what's the actual question you are trying to ask and how does it relates to AdGuard?

**Comment by techIndia-hacker** [comment]
The certificate is getting placed in system store but Google apps don't trust it

**Comment by techIndia-hacker** [comment]
After recent Google Play Store update, I can't download app updates due to certificate non trust issue in OnePlus phone but on the other hand samsung phone is working normally without certificate issues. Both running Android 10. Only the process of placing adguard certificate in trusted store is different.

Note:- I have disabled https opportunistic in low level settings as it opens gate for trackers on both devices.

**Comment by TheHasagi** [comment]
@techIndia-hacker 

To troubleshoot this issue, we need to get additional logs.

Here's what we need you to do:

1. Collect the debug log as it's [explained here](https://kb.adguard.com/en/android/solving-problems/log#debug).
2. Remember the exact time when the issue was reproduced, we will need it to find the corresponding records in the log file.
3. Additionally, capture the system bug report after reproducing the issue. [Here's the instruction](https://developer.android.com/studio/debug/bug-report).
4. Send both files to `devteam@adguard.com`. Mention the Github issue number and the exact time when the issue was reproduced.

**Comment by techIndia-hacker** [comment]
Mailing to devteam within 5 minutes. Please check and let me know

**Comment by admitrevskiy** [comment]
@techIndia-hacker hi!
This issue is not reproducable on our side (One Plus 6T) and there is nothing remarkable in the logs: GPlay constantly rejects our certificate moved with Magisk.
You can try to remove all AG certs from all the storages, generate the new one and try to move it again.
Anyway it looks like an issue for Magisk team, not for us. Please let me know if they ask you for extra information, we will try to provide them all necessary info to fix this issue.

**Comment by techIndia-hacker** [comment]
Ok ur OnePlus 6T Google Play rejects certificate when moved with magisk right?
But if u move the certificate with the "move anyway" option in adguard, whats the result?
In my Samsung A70 Google Play accepts adguard certificate and works smoothly. This is what I am reporting.
But in my OnePlus 8 this can't be done as adguard fails to move the certificate. And when moved using magisk adguard certificate is not trusted system wide

**Comment by admitrevskiy** [comment]
No, as I've mentioned above this issue is not reproducable on our side: OnePlus 6T Google Play accepts certificate moved with Magisk.
Not all versions of Magisk allow us to move the certificate using our tools (aka "Move anyway" option). That's why we recommend to use Magisk MoveCerts module. Moreover we cannot control this process since it's Magisk job, not ours.

**Comment by techIndia-hacker** [comment]
Please share the code at least do I can diagnose. I am a developer too

**Comment by techIndia-hacker** [comment]
I have solved this issue. After much sleepless nights I finally was able to kill Google SSL pinning. If adguard developers think that this is a important thing, they can contact me
